### PR TITLE
[AMORO-2692] Support proxy user in terminal

### LINKED ISSUE #2692: [Feature]: Support proxy users in spark terminal
### Description

We should support proxy users in the spark terminal to improve multi-tenancy experience.the 

### Use case/motivation

_No response_

### Describe the solution

_No response_

### Subtasks

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by link3280**
## Why are the changes needed?

Close #2692.

## Brief change log

- Add proxy user parameter to the REST API.
- Support impersonation in spark terminal.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by xieyi888** [comment]
I‘m looking forwared to this PR!  But I have some questions
It seems ams-server did not include authorization and authentication, How can we use this function on ams terminal?
Do we consider support proxy on spark engine?

**Comment by link3280** [comment]
@xieyi888 Thanks for your input! 

> It seems ams-server did not include authorization and authentication, How can we use this function on ams terminal?

Do you mean the kerberos authentication? Ideally, the terminal would use the UGI in TableMetaStore.RuntimeContext, but in fact the UGI cannot cover the usages like Iceberg async tasks. This PR mainly makes the spark user right. For example, the table created would have the proxy user as its owner.

WRT authorization, I think that's the catalog's job and not relevant to the amoro terminal.

> Do we consider support proxy on spark engine?

I think it's doable via Kyuubi. But I haven't got the chance to verify it yet.

**Comment by baiyangtx** [comment]
@link3280 From your PR, it seems that you are calling terminal by REST API?  so that you can pass the proxyUser from http parameters. But this is not working for user executing SQL via WebBrowser.

**Comment by link3280** [comment]
> @link3280 From your PR, it seems that you are calling terminal by REST API? so that you can pass the proxyUser from http parameters. But this is not working for user executing SQL via WebBrowser.

That's right. This PR doesn't involve web UI changes.

**Comment by XBaith** [comment]
> I think it's doable via Kyuubi. But I haven't got the chance to verify it yet.

Hi @link3280, Can you explain more details how to execute SQL statements with configured user?

File: ams/server/src/main/java/com/netease/arctic/server/terminal/TerminalSessionContext.java [hunk 1]
```diff
@@ -278,7 +298,9 @@ boolean executeStatement(TerminalSession session, String statement, int lineNo)
       TerminalSession.ResultSet rs = null;
       long begin = System.currentTimeMillis();
       try {
-        rs = session.executeStatement(catalog, statement);
+        rs =
+            tableMetaStore.doAsImpersonating(
+                proxyUser, () -> session.executeStatement(catalog, statement));
```
[the remark this answers is not in the document]
Comment (zhoujinsong) [reply ams/server/src/main/java/com/netease/arctic/server/terminal/TerminalSessionContext.java:303]: +1 for this suggestion

**Pull request closed without merging**